NGitLab/Impl/PackageClient.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using NGitLab.Models;
+
+namespace NGitLab.Impl
+{
+    public class PackageClient : IPackageClient
+    {
+        private const string PublishPackageUrl = "/projects/{0}/packages/generic/{1}/{2}/{3}?status={4}&select=package_file";
+        private const string GetPackagesUrl = "/projects/{0}/packages";
+        private const string GetPackageUrl = "/projects/{0}/packages/{1}";
+
+        private readonly API _api;
+
+        public PackageClient(API api)
+        {
+            _api = api;
+        }
+
+        public Task<Package> PublishAsync(int projectId, PackagePublish packagePublish, CancellationToken cancellationToken = default)
```

Comment:
Based on the url, I think the method name should be `PublishGenericPackageAsync`

```suggestion
        public Task<Package> PublishGenericPackageAsync(int projectId, PackagePublish packagePublish, CancellationToken cancellationToken = default)
```